src/hidimstat/base_perturbation.py

```diff
@@ -0,0 +1,178 @@
+"""Base class for model agnostic variable importance measures based on perturbation."""
+
+import numpy as np
+import pandas as pd
+from joblib import Parallel, delayed
+from sklearn.base import BaseEstimator, check_is_fitted
+from sklearn.metrics import root_mean_squared_error
+
+from hidimstat.utils import _check_vim_predict_method
+
+
+class BasePerturbation(BaseEstimator):
+    def __init__(
+        self,
+        estimator,
+        loss: callable = root_mean_squared_error,
+        n_permutations: int = 50,
+        method: str = "predict",
+        n_jobs: int = 1,
+    ):
+        """
+        Base class for model-agnostic variable importance measures based on
+        perturbation.
+
+        Parameters
+        ----------
+        estimator : object
```

Comment:
If you can replace the description of the estimator with the following:
" estimator : sklearn compatible estimator, optional
            The estimator to use for the prediction."
It's the description of the estimator in CPI.